ffmpeg/ffmpeg_test.go

```diff
@@ -575,3 +575,311 @@ func TestTranscoder_StatisticsAspectRatio(t *testing.T) {
 		t.Error(fmt.Errorf("Results did not match: %v ", r))
 	}
 }
+
```

Comment:
I wonder if the things might be clearer during unit testing if this `ffmpeg_test.go` were broken down into smaller files.....

Reply:
How should we split the file up? The segmenter- related tests could probably be put into its own file, but that's only ~100LOC or so. Pretty much everything else exercises various aspects of the transcoder. (That reminds me, these tests should probably be prefixed with `Transcoder_` to follow the naming convention in the rest of this file.)

There is some boilerplate here that I've been meaning to tidy up separately. Eg, basically all the shell tests start out with this line `set -eux ; cd $0` which can probably be extracted into a preamble that's automatically prefixed before `run`. Also the test file from `../transcoder/test.ts` is usually copied to the tmpdir and shortened to ~1 second; this can probably be done during the `setupTest` phase as well.

Using the testify package would probably save a few lines as well, but we don't really use that many assertions here.

Reply:
About the boilerplate stuff, it sounds like a good idea, and it should potentially make writing future tests easier.  Be sure to keep in mind that readability is the most important part of the tests - other developers will look at the unit tests in an effort to understand how each function works.  It sounds like extracting those few commands into a preamble would certainly help with this, but let's do it in a separate PR after this one is merged.

As for splitting the file up, it should also be handled separately after this PR is merged.  A nice thing about having many smaller test files rather than fewer larger test files is that when there are failures, they are easier to locate, due to the fact that the filename is displayed when a test fails.

100 LOC is not too small for a separate file...  If everything else is related to the transcoder, maybe there is another way to break it up.    If there isn't and we have to keep all the transcoder tests together, it's not the end of the world.... This was just a suggestion.